src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -262,7 +257,10 @@ impl<S: StateStore> StorageTable<S> {
             )
             .await?
         {
-            let full_row = self.row_deserializer.deserialize(value).map_err(err)?;
+            let full_row = self
+                .row_deserializer
+                .deserialize(value)
+                .map_err(StorageError::deserialize_row_error)?;
```

Comment:
Can we `impl From<ValueEncodingError> for StorageError` to remove this explicit `map_err`?